deploy/crds/openebs.io_blockdeviceclaims_crd.yaml

```diff
@@ -0,0 +1,159 @@
+                  from which a BD should be selected for this claim. It can include
+                  nodename, failure domain etc.
+                properties:
+                  hostName:
+                    description: HostName represents the hostname of the Kubernetes
+                      node resource where the BD should be present
+                    type: string
+                  nodeName:
+                    description: NodeName represents the name of the Kubernetes node
+                      resource where the BD should be present
+                    type: string
+                type: object
+              deviceClaimDetails:
+                description: Details of the device to be claimed
+                properties:
+                  allowPartition:
+                    description: AllowPartition represents whether to claim a full
+                      block device or a device that is a partition
+                    type: boolean
+                  blockVolumeMode:
+                    description: 'BlockVolumeMode represents whether to claim a device
+                      in Block mode or Filesystem mode. These are use cases of BlockVolumeMode:
+                      1) Not specified: VolumeMode check will not be effective 2)
+                      VolumeModeBlock: BD should not have any filesystem or mountpoint
+                      3) VolumeModeFileSystem: BD should have a filesystem and mountpoint.
+                      If DeviceFormat is    specified then the format should match
+                      with the FSType in BD'
+                    type: string
+                  formatType:
+                    description: Format of the device required, eg:ext4, xfs
+                    type: string
+                type: object
+              deviceType:
+                description: DeviceType represents the type of drive like SSD, HDD
+                  etc.,
+                type: string
+              hostName:
+                description: Node name from where blockdevice has to be claimed. To
+                  be deprecated. Use NodeAttributes.HostName instead
+                type: string
+              resources:
+                description: Resources will help with placing claims on Capacity,
+                  IOPS
+                properties:
+                  requests:
+                    additionalProperties:
+                      anyOf:
+                      - type: integer
+                      - type: string
+                      pattern: ^(\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))(([KMGTPE]i)|[numkMGTPE]|([eE](\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))))?$
+                      x-kubernetes-int-or-string: true
+                    description: 'Requests describes the minimum resources required.
+                      eg: if storage resource of 10G is requested minimum capacity
+                      of 10G should be available'
+                    type: object
+                required:
```

Comment:
Is there a way to say the list of keys that are supported in the map, like we need only `storage` key in the map. All other keys should be invalid.